CIME/case/case_setup.py

```diff
@@ -141,12 +141,30 @@ def _create_macros_cmake(
     ###############################################################################
     if not os.path.isfile(os.path.join(caseroot, "Macros.cmake")):
         safe_copy(os.path.join(cmake_macros_dir, "Macros.cmake"), caseroot)
-    if not os.path.exists(os.path.join(caseroot, "cmake_macros")):
-        shutil.copytree(cmake_macros_dir, case_cmake_path)
 
-    copy_depends_files(
-        mach_obj.get_machine_name(), mach_obj.machines_dir, caseroot, compiler
-    )
+    mach = mach_obj.get_machine_name()
+    if Config.instance().copy_all_cmake_macros:
+        if not os.path.exists(case_cmake_path):
+            shutil.copytree(cmake_macros_dir, case_cmake_path)
+    else:
+        if not os.path.exists(case_cmake_path):
+            os.mkdir(case_cmake_path)
+
+        # This impl is coupled to contents of Macros.cmake
+        macros = [
+            "universal.cmake",
```

Comment:
I think that you should include cesm.  To do that you just need to add a couple more files:
${OS}.cmake
${COMPILER}_${OS}.cmake

Reply:
our Macros.cmake file is generated in cime/CIME/build.py

Reply:
Yes - I just tried a build with this PR and it works fine.

Reply:
@jedwards4b , great. So do you approve of this PR now?